src/main/ruby/core/truffle/cext.rb

```diff
@@ -181,6 +181,17 @@ def fd
   end
 
   class RStringPtr
+    RSTRING_CACHE = {}
+    RSTRING_CACHE_MUTEX = Mutex.new
+
+    private_class_method :new
+
+    def self.get(string)
+      RSTRING_CACHE_MUTEX.synchronize do
+        RSTRING_CACHE.fetch(string.object_id) { |key| RSTRING_CACHE[key] = new(string) }
```

Comment:
A compare_by_identity Hash could be used here I think.

Reply:
Agreed.